Description: Update for nix 0.23
 The nix crate revamped it's error handling, this patch adjusts x11rb
 in a minimal manner so it will build with the new version of the nix
 crate in a minimal manner.
 
 Upstream also upgraded nix in version 0.9, but they did some cleanup
 at the same time, so their patch is larger than this one.
Author: Peter Michael Green <plugwash@debian.org>


--- rust-x11rb-0.8.1.orig/Cargo.toml
+++ rust-x11rb-0.8.1/Cargo.toml
@@ -97,7 +97,7 @@ xtest = []
 xv = ["shm"]
 xvmc = ["xv"]
 [target."cfg(unix)".dependencies.nix]
-version = "0.20"
+version = "0.23"
 #[target."cfg(windows)".dependencies.winapi]
 #version = "0.3"
 #features = ["winsock2"]
--- rust-x11rb-0.8.1.orig/examples/xclock_utc.rs
+++ rust-x11rb-0.8.1/examples/xclock_utc.rs
@@ -195,7 +195,7 @@ fn poll_with_timeout(
                 }
             }
             // try again
-            Err(nix::Error::Sys(nix::errno::Errno::EINTR)) => {}
+            Err(nix::Error::EINTR) => {}
             Err(e) => return Err(e.into()),
         }
         if start_instant.elapsed() >= timeout {
--- rust-x11rb-0.8.1.orig/src/rust_connection/stream.rs
+++ rust-x11rb-0.8.1/src/rust_connection/stream.rs
@@ -355,7 +355,7 @@ fn do_write(
             match sendmsg(fd, iov, cmsgs, flags, addr) {
                 Ok(n) => return Ok(n),
                 // try again
-                Err(nix::Error::Sys(nix::errno::Errno::EINTR)) => {}
+                Err(nix::Error::EINTR) => {}
                 Err(e) => return Err(nix_error_to_io(e)),
             }
         }
@@ -396,7 +396,7 @@ impl Stream for DefaultStream {
             loop {
                 match poll(&mut poll_fds, -1) {
                     Ok(_) => break,
-                    Err(nix::Error::Sys(Errno::EINTR)) => {}
+                    Err(nix::Error::EINTR) => {}
                     Err(e) => return Err(nix_error_to_io(e)),
                 }
             }
@@ -445,7 +445,7 @@ impl Stream for DefaultStream {
                 match recvmsg(fd, &iov[..], Some(&mut cmsg), MsgFlags::empty()) {
                     Ok(msg) => break msg,
                     // try again
-                    Err(nix::Error::Sys(nix::errno::Errno::EINTR)) => {}
+                    Err(nix::Error::EINTR) => {}
                     Err(e) => return Err(nix_error_to_io(e)),
                 }
             };
--- rust-x11rb-0.8.1.orig/src/utils.rs
+++ rust-x11rb-0.8.1/src/utils.rs
@@ -109,13 +109,7 @@ pub use unsafe_code::CSlice;
 
 #[cfg(unix)]
 pub(crate) fn nix_error_to_io(e: nix::Error) -> std::io::Error {
-    match e {
-        nix::Error::Sys(errno) => errno.into(),
-        nix::Error::InvalidPath | nix::Error::InvalidUtf8 => {
-            std::io::Error::new(std::io::ErrorKind::InvalidInput, e)
-        }
-        nix::Error::UnsupportedOperation => std::io::Error::new(std::io::ErrorKind::Other, e),
-    }
+    e.into()
 }
 
 #[cfg(unix)]
